.github/workflows/isolated_migrations.yml

```diff
@@ -0,0 +1,69 @@
+# GitHub action that checks:
+# - that the migrations and the related code changes are not in the same pull request
+# - that the schema.rb and data_schema.rb files are included in the pull request that introduces the corresponding migration
+# Based on https://github.com/marketplace/actions/changed-files
+
+name: Isolated Migrations
+
+on:
+  pull_request
+
+jobs:
+  check_changed_files:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Group files that have changed
+        id: changed-files-yaml
+        uses: tj-actions/changed-files@v39
+        with:
+          files_yaml: |
+            migrations:
+              - src/api/db/**
+            not_migrations:
+              - '!src/api/db/**'
+            schema_migrations:
+              - src/api/db/migrate/**
+            data_migrations:
+              - src/api/db/data/**
+            schema:
+              - src/api/db/schema.rb
+            data_schema:
+              - src/api/db/data_schema.rb
+
+      - name: List the files related to migrations that this PR changes
```

Comment:
Is there a reasoning for having this step here? Same for the other steps listing files. What is the benefit in having such steps?

Reply:
If the check fails, it's helpful to see what are the files included in the PR to understand why. I thought it was better to see them in the output than forcing the user to move back to the PR to see the _Files Changed_.

Moreover, while we fine-tune this GH action is good to see what is the action considering as migration-related files or non-migration-related files.

Reply:
Good points :+1: